src/lib/transaction_logic/zkapp_command_logic.ml

```diff
@@ -1142,6 +1162,14 @@ module Make (Inputs : Inputs_intf) = struct
         (Account_update.token_id account_update)
         (a, inclusion_proof)
     in
+    let matching_verification_key_hashes =
+      Inputs.Bool.(
+        (not (Account_update.is_proved account_update))
+        ||| Verification_key_hash.equal
+              (Account.verification_key_hash a)
+              (Account_update.verification_key_hash account_update))
+    in
+    assert_ ~pos:__POS__ matching_verification_key_hashes ;
```

Comment:
Can you call `Local_state.add_check` instead of raising. We also need a new failure to identify this error

Reply:
Done.